crates/sui-core/src/execution_cache/unit_tests/writeback_cache_tests.rs

```diff
@@ -0,0 +1,515 @@
+            let LiveObject::Normal(o) = o else {
+                panic!("expected normal object")
+            };
+            let id = o.id();
+            // genesis objects are not managed by Scenario, ignore them
+            if reverse_id_map.get(&id).is_some() {
+                self.objects.insert(id, o);
+            }
+        });
+    }
+
+    fn assert_live(&self, short_ids: &[u32]) {
+        for short_id in short_ids {
+            let id = self.id_map.get(short_id).expect("no such object");
+            let expected = self.objects.get(id).expect("no such object");
+            let version = expected.version();
+            assert_eq!(
+                self.cache()
+                    .get_object_by_key(id, version)
+                    .unwrap()
+                    .unwrap(),
+                *expected
+            );
+            assert_eq!(
+                self.cache().get_object(&expected.id()).unwrap().unwrap(),
+                *expected
+            );
+            // TODO: enable after lock caching is implemented
+            // assert!(!self
+            //  .cache()
+            //  .get_lock(expected.compute_object_reference(), 1)
+            //  .unwrap()
+            //  .is_locked());
+        }
+    }
+
+    fn assert_received(&self, short_ids: &[u32]) {
+        for short_id in short_ids {
+            let id = self.id_map.get(short_id).expect("no such object");
+            let object = self.objects.get(id).expect("no such object");
+            assert_eq!(
+                self.cache()
+                    .get_object_by_key(id, object.version())
+                    .unwrap()
+                    .unwrap(),
+                *object
+            );
+            assert!(self
+                .cache()
+                .have_received_object_at_version(id, object.version(), 1)
+                .unwrap());
+        }
+    }
+
+    fn assert_not_exists(&self, short_ids: &[u32]) {
+        for short_id in short_ids {
+            let id = self.id_map.get(short_id).expect("no such id");
+
+            assert!(
+                self.cache().get_object(id).unwrap().is_none(),
+                "object exists in cache"
+            );
+        }
+    }
+
+    fn obj_id(&self, short_id: u32) -> ObjectID {
+        *self.id_map.get(&short_id).expect("no such id")
+    }
+}
+
+#[tokio::test]
+async fn test_uncommitted() {
+    telemetry_subscribers::init_for_testing();
+    Scenario::iterate(|mut s| async move {
+        s.with_created(&[1, 2]);
+        s.do_tx().await;
+
+        s.assert_live(&[1, 2]);
+        s.reset_cache();
+        s.assert_not_exists(&[1, 2]);
+    })
+    .await;
+}
+
+#[tokio::test]
+async fn test_committed() {
+    telemetry_subscribers::init_for_testing();
+    Scenario::iterate(|mut s| async move {
+        s.with_created(&[1, 2]);
+        let tx = s.do_tx().await;
+
+        s.assert_live(&[1, 2]);
+        s.cache()
+            .commit_transaction_outputs(1, &tx)
```

Comment:
Maybe consider also check internal state of dirty, cache, and DB after commit_transaction_outputs(). Seems a critical function to be tested thoroughly. (If worth doing, maybe just add a TODO here for now and do it after)

Reply:
added a test which does this